### PR TITLE
Automation updates of `metadata` table 

### PR DESCRIPTION
**Pull request by bogdasar1985**
Issue https://github.com/newsboat/newsboat/issues/2528

### CODE REVIEW

File: src/cache.cpp [hunk 1]
```diff
@@ -435,7 +432,15 @@ void Cache::populate_tables()
...
 			patch_version.major,
 			patch_version.minor);
+
+		std::string update_metadata_query = "UPDATE metadata SET db_schema_version_major = " +
+			std::to_string(patch_version.major) + ", db_schema_version_minor = " +
+			std::to_string(patch_version.minor) + ";";
+
 		for (const auto& query : patches_it->second) {
+			if (patch_version > SchemaVersion{2, 11}) {
+				run_sql_nothrow(update_metadata_query);
```
Comment (dennisschagt) [inline src/cache.cpp:442]: I think this only needs to be run once per set of SQL queries, so just outside of this inner for-loop.

**Review by dennisschagt — COMMENTED** [review]
Nice improvement!

File: src/cache.cpp (hunk 1 above)
Comment (bogdasar1985) [reply src/cache.cpp:442]: @dennisschagt that's right

**Review by dennisschagt — APPROVED** [review]
Thanks, LGTM

**Pull request merged**